### ISSUE TITLE
v3 Alpha: Testing & Feedback (Multi-Accounts, ...)

### ISSUE THREAD

**Issue by timche** [timche/gmail-desktop#177]
We are currently working on the next major version of Gmail Desktop which includes the long-awaited multi-accounts support. The reason for a major version is that Gmail Desktop will be re-written for most parts in order to make multi-accounts a great experience without suffering any existing features. Another reason is that Gmail Desktop will be renamed to Mazy (hope you like the name), so it must be re-installed and can't be updated automatically from v2 to v3.

Before releasing v3, I'm looking for alpha testers that can provide feedback to ensure everything works as intended and performance is alright.

**Current v3 changes:**
- Support multi-account (#25)
- Ability to change release channel (`latest`, `beta`, `alpha`)
- Rename project to Mazy or Mailzy

**Latest v3 alpha release:** 
- https://github.com/timche/gmail-desktop/releases/tag/v3.0.0-alpha.1
- Please set the release channel to `Alpha` at `Settings` → `Updates` → `Release Channel` in order to receive future alpha updates.

**How to provide feedback:**
```
- OS:
- Version:

<Feedback/Bug Report>
```

**Known bugs/issues:**
- macOS
  - [Testing in progress]
- Linux
  - [Testing in progress]
- Windows
  - [Testing in progress]

<details>
<summary><b>Previews:</b></summary>

![Screen Shot 2021-03-14 at 22 07 37](https://user-images.githubusercontent.com/8075074/111085959-b9054280-8511-11eb-839c-f8f73a0af46f.png)

![Screen Shot 2021-03-14 at 22 07 46](https://user-images.githubusercontent.com/8075074/111085963-bacf0600-8511-11eb-8310-5b160f2092b8.png)

![Kapture 2021-03-14 at 22 10 33](https://user-images.githubusercontent.com/8075074/111086127-80b23400-8512-11eb-9435-5fc755466f1a.gif)

![Kapture 2021-03-14 at 22 09 17](https://user-images.githubusercontent.com/8075074/111086143-8a3b9c00-8512-11eb-94ed-2016806e9d46.gif)

![Kapture 2021-03-14 at 22 11 28](https://user-images.githubusercontent.com/8075074/111086144-8dcf2300-8512-11eb-894a-440c86b512cb.gif)
</details>

**Comment by cyfrost** [comment]
Hey @timche, great initiative! I'm happy to test the alpha once it's out. Fedora, Arch Linux, Ubuntu, and Windows 10 are the ones I can lay my hands on for testing. Let me know when the alpha is ready! :)

**Comment by zordym** [comment]
Hello @timche, 

I can test your application.

My setup: Mac os catalina

**Comment by zzpxyx** [comment]
@timche Hopefully I'm not late for the game. I have access to Arch Linux, macOS Catalina, and Windows 10.

**Comment by Cantello** [comment]
Would it also be possible with this to switch to other Google services (thinking of calendar & Docs)?

**Comment by timche** [comment]
> Would it also be possible with this to switch to other Google services (thinking of calendar & Docs)?

There are some apps out there that support multiple Google services, but I haven't tried them yet. My primary goal with Gmail Desktop is to focus on Gmail and eventually add other Google services in the future.

**Comment by cipriancraciun** [comment]
I can test it on OpenSUSE Tumbleweed (I don't need any special package except the Linux x86-64 `.tar.bz` that you already are providing.)

**Comment by cipriancraciun** [comment]
> There are some apps out there that support multiple Google services, but I haven't tried them yet. My primary goal with Gmail Desktop is to focus on Gmail and eventually add other Google services in the future.

@timche Please keep this project GMail focused. :)

As you've mentioned there are other projects out there that try to do all this, but given that they don't have any focus on any of these "targeted sites", they aren't best at anything.

Meanwhile this project already does provide useful features for GMail, especially in the UI department by cleaning up some GMail artifacts.

**Comment by akamizi** [comment]
OS: Elementary OS
I have various normal and business accounts
I am currently using this:
https://github.com/openWMail/openWMail

**Comment by axllent** [comment]
I'd love to test this as well once it is ready, as I have 5 separate Gmail accounts (personal & business) I need to check. Two machines, Ubuntu 16 & 20 (Unity on both).

**Comment by Chriscbr** [comment]
I'd be happy to test this feature (I have about three Gmail accounts I check on at the moment). I can test on macOS and Windows 10 with my currently available hardware. 😄

**Comment by pedrodonkey** [comment]
✋ Available to test if this is still in the works. Running macOS Catalina 10.15.6 (19G2021).

**Comment by timche** [comment]
It has been a while and I'm just re-opening this issue, because I'm happy to announce that v3 is now finally coming! I hope that most of you are still sticking around and are still interested to be alpha testers.

I'll release the first alpha version very soon, but in the meantime please enjoy some previews of what's coming:

![Screen Shot 2021-03-14 at 22 07 37](https://user-images.githubusercontent.com/8075074/111085959-b9054280-8511-11eb-839c-f8f73a0af46f.png)

![Screen Shot 2021-03-14 at 22 07 46](https://user-images.githubusercontent.com/8075074/111085963-bacf0600-8511-11eb-8310-5b160f2092b8.png)

![Kapture 2021-03-14 at 22 10 33](https://user-images.githubusercontent.com/8075074/111086127-80b23400-8512-11eb-9435-5fc755466f1a.gif)

![Kapture 2021-03-14 at 22 09 17](https://user-images.githubusercontent.com/8075074/111086143-8a3b9c00-8512-11eb-94ed-2016806e9d46.gif)

![Kapture 2021-03-14 at 22 11 28](https://user-images.githubusercontent.com/8075074/111086144-8dcf2300-8512-11eb-894a-440c86b512cb.gif)

CC @duong-le @brahimmouhamou @floriangbh @gurumark @bsramin @mskelton

**Comment by timche** [comment]
The first alpha version is released!

https://github.com/timche/gmail-desktop/releases/tag/v3.0.0-alpha.1

**Comment by bsramin** [comment]
Great Job! 👏

**Comment by duong-le** [comment]
![image](https://user-images.githubusercontent.com/28813618/111118114-f661e280-859a-11eb-98ae-97dbb73fb50e.png)
So I did set up 2 accounts, enabled `Hide footer` and this is the result I get from the 1st account. The second account is perfectly fine, any idea?

**Comment by timche** [comment]
@duong-le Unfortunately it's difficult to judge where why it's cut off. I'll provide an option to open the dev tools for Gmail so you can have a look yourself.

**Comment by brahimmouhamou** [comment]
hi @timche,

Thanks for this release, eager to use it!

I'm having a problem adding my Gmail account, I always get the message "This browser or app may not be secure".
So I was following the troubleshooting steps, https://github.com/timche/gmail-desktop#troubleshooting. I added the latest browser:

`"customUserAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 11_2_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.90 Safari/537.36",`

When adding my **Google for Work address, everything works**, but when adding my **Gmail account, it fails** and I keep getting the message "This browser or app may not be secure".

My OS: **macOS Catalina 10.15.6**

Should I create a separate issue for this?
Can I try something else to fix this?

Kind regards,

**Comment by timche** [comment]
@brahimmouhamou were you getting the pop up to let Gmail Desktop fix it automatically like described in the [troubleshooting section](https://github.com/timche/gmail-desktop#i-cant-sign-in-this-browser-or-app-may-not-be-secure) when shown the message "This browser or app may not be secure"? 

![](https://github.com/timche/gmail-desktop/raw/main/media/sign-in-fix-dialog.png)

**Comment by brahimmouhamou** [comment]
@timche sorry, it's fixed!

I did the popup to auto fix it and tried to auto fix it before but I kept having the same problem.
But now I tried it again and it works now. Only difference that know is that I already have an existing account (my Google for work) or some kind of caching (?).

**Comment by timche** [comment]
> @timche sorry, it's fixed!
> 
> I did the popup to auto fix it and tried to auto fix it before but I kept having the same problem.
> But now I tried it again and it works now. Only difference that know is that I already have an existing account (my Google for work) or some kind of caching (?).

@brahimmouhamou each account is having their own session storage, so nothing is shared. The user agent is a global setting for all accounts. You can try to reproduce it with `Settings` → `Advanced` → `User Agent` → `Remove Custom User Agent` and log in again in the affected account.

**Comment by timche** [comment]
> Seems alpha release is not signed and cannot be launched on Mac OS when installed manually through a .dmg download either.

@pedrodonkey I've rebuilt the v3.0.0-alpha.1 release with updated macOS certificates. Can you please check again and provide screenshots of the warning/error message if necessary?
